modules/reactor/src/meshgenerators/RevolveGenerator.C

```diff
@@ -0,0 +1,2340 @@
+                                                            _elem_integers_swaps,
+                                                            _elem_integers_swap_pairs);
+  }
+  catch (const MooseException & e)
+  {
+    paramError("elem_integers_swaps", e.what());
+  }
+}
+
+std::unique_ptr<MeshBase>
+RevolveGenerator::generate()
+{
+  // Note: Inspired by AdvancedExtruderGenerator::generate()
+
+  auto mesh = buildMeshBaseObject();
+  mesh->set_mesh_dimension(_input->mesh_dimension() + 1);
+
+  // Check if the element integer names are existent in the input mesh.
+  for (unsigned int i = 0; i < _elem_integer_names_to_swap.size(); i++)
+    if (_input->has_elem_integer(_elem_integer_names_to_swap[i]))
+      _elem_integer_indices_to_swap.push_back(
+          _input->get_elem_integer_index(_elem_integer_names_to_swap[i]));
+    else
+      paramError("elem_integer_names_to_swap",
+                 "Element ",
+                 i + 1,
+                 " of 'elem_integer_names_to_swap' in is not a valid extra element integer of the "
+                 "input mesh.");
+
+  // prepare for transferring extra element integers from original mesh to the revolved mesh.
+  const unsigned int num_extra_elem_integers = _input->n_elem_integers();
+  std::vector<std::string> id_names;
+
+  for (unsigned int i = 0; i < num_extra_elem_integers; i++)
+  {
+    id_names.push_back(_input->get_elem_integer_name(i));
+    if (!mesh->has_elem_integer(id_names[i]))
+      mesh->add_elem_integer(id_names[i]);
+  }
+
+  // retrieve subdomain/sideset/nodeset name maps
+  const auto & input_subdomain_map = _input->get_subdomain_name_map();
+  const auto & input_sideset_map = _input->get_boundary_info().get_sideset_name_map();
+  const auto & input_nodeset_map = _input->get_boundary_info().get_nodeset_name_map();
+
+  std::unique_ptr<MeshBase> input = std::move(_input);
+
+  // If we're using a distributed mesh... then make sure we don't have any remote elements hanging
+  // around
+  if (!input->is_serial())
+    mesh->delete_remote_elements();
+
+  // Subdomain IDs for on-axis elements must be new
+  std::set<subdomain_id_type> subdomain_ids_set;
+  input->subdomain_ids(subdomain_ids_set);
+  const subdomain_id_type max_subdomain_id = *subdomain_ids_set.rbegin();
+  const subdomain_id_type tri_to_pyramid_subdomain_id_shift =
+      std::max((int)max_subdomain_id, 1) + 1;
+  const subdomain_id_type tri_to_tet_subdomain_id_shift =
+      std::max((int)max_subdomain_id, 1) * 2 + 1;
+  const subdomain_id_type quad_to_prism_subdomain_id_shift = std::max((int)max_subdomain_id, 1) + 1;
+  const subdomain_id_type quad_to_pyramid_subdomain_id_shift =
+      std::max((int)max_subdomain_id, 1) * 2 + 1;
+  const subdomain_id_type quad_to_hi_pyramid_subdomain_id_shift =
+      std::max((int)max_subdomain_id, 1) * 3 + 1;
+  const subdomain_id_type edge_to_tri_subdomain_id_shift = std::max((int)max_subdomain_id, 1) + 1;
```

Comment:
isnt this going to overlap with quad_to_prism ?

Reply:
If the original element types are different (quad vs. edge), the original subdomain ids are different. In that case, it is okay to have the same shift.